libarchive/archive.h

```diff
@@ -120,6 +120,8 @@ typedef ssize_t la_ssize_t;
 #   define __LA_DECL	__declspec(dllimport)
 #  endif
 # endif
+#elif defined __LIBARCHIVE_ENABLE_VISIBILITY
+#  define __LA_DECL __attribute__((visibility("default")))
 #else
 /* Static libraries or non-Windows needs no special declaration. */
 # define __LA_DECL
```

Comment:
`__LA_DECL` is now handling several different issues, so I'm tempted to refactor `__LA_DECL` as the composite of several different attributes:
```
#if ... windows ...
#define __LA_DECL_IMPORT  ...
#endif

#if ... windows ...
#define __LA_DECL_EXPORT ...
#endif

#if defined(__LIBARCHIVE_ENABLE_VISIBILITY)
#define __LA_DECL_VISIBILITY __attribute__((visibility("default")))
#else
#define __LA_DECL_VISIBILITY
#endif 

#define __LA_DECL  __LA_DECL_VISIBILITY __LA_DECL_IMPORT __LA_DECL_EXPORT
```

Reply:
Do you suggest to do it in this PR?

Reply:
No.  We can do that refactoring in a separate PR if you prefer.